execution/evm/vm.go

```diff
@@ -871,6 +913,8 @@ func (vm *VM) call(caller acm.Account, callee acm.MutableAccount, code, input []
 			} else {
 				newAccount.SetCode(ret) // Set the code (ret need not be copied as per Call contract)
 				stack.Push(newAccount.Address().Word256())
+				emptyBuffer := []byte{}
+				vm.setReturnDataBuffer(emptyBuffer)
```

Comment:
When there is an exception After line 912 we need:

```go
vm.returnData = ret
```
since the EIP states:
> As an exception, CREATE and CREATE2 are considered to return the empty buffer in the success case and the failure data in the failure case

I also note that while I think you are following the spec here, geth and eth-cpp appear to always return what CREATE returns. Will mull this over.

However we should be clearing the buffer when we enter `CREATE` on line 879. Also there is no need to create an empty buffer, just use nil:

```go
vm.returnData = nil
```

And remove the clearing of the buffer where you currently have it. Either we will have set the buffer to ret on failure or it will be empty anyway.